lib/rubocop/cop/layout/space_around_block_parameters.rb

```diff
@@ -116,18 +117,30 @@ def check_opening_pipe_space(args, opening_pipe)
                          'Extra space before first')
         end
 
-        def check_closing_pipe_space(args, closing_pipe)
+        def check_closing_pipe_space(arguments, closing_pipe)
+          args = arguments.children
+
           last         = args.last.source_range
-          last_end_pos = last_end_pos_inside_pipes(last.end_pos)
+          last_end_pos = last_end_pos_inside_pipes(arguments, last.end_pos)
 
           check_space(last_end_pos, closing_pipe.begin_pos, last,
                       'after last block parameter')
           check_no_space(last_end_pos + 1, closing_pipe.begin_pos,
                          'Extra space after last')
         end
 
-        def last_end_pos_inside_pipes(pos)
-          processed_source.buffer.source[pos] == ',' ? pos + 1 : pos
+        def last_end_pos_inside_pipes(arguments, pos)
+          num = pos - arguments.source_range.to_a.first
+          trailing_comma_index = arguments.source[num..-1].index(',')
```

Comment:
There must be a smarter way to do this :-)

Reply:
I don't think so; `parser` doesn't give out any info about the commas